src/third_party/wiredtiger/src/txn/txn_ckpt.c

```diff
@@ -1335,8 +1335,17 @@ __checkpoint_lock_dirty_tree(WT_SESSION_IMPL *session,
 	WT_ASSERT(session, !need_tracking ||
 	    WT_IS_METADATA(dhandle) || WT_META_TRACKING(session));
 
-	/* Get the list of checkpoints for this file. */
-	WT_RET(__wt_meta_ckptlist_get(session, dhandle->name, &ckptbase));
+	/*
+	 * Get the list of checkpoints for this file.
+	 * Use the custom checkpoint metadata parser if configured.
+	 */
+	if (F_ISSET(conn, WT_CONN_CKPT_METADATA_FAST_PARSER)) {
```

Comment:
I think we could do this inside of __wt_meta_ckptlist_get() and avoid duplicating the beginning and end of that function.
If the issue is to get conn in that function, S2C should work.

Reply:
Hm ok. I was trying to keep the original code the same, so if anything goes wrong and we flip the flag it's as good as a revert. I can refactor some of this and try to keep it low risk.

Reply:
I see. The drawback is if we patch other versions, we have to check if anything else changed.

I think you could do something light in __wt_meta_ckplist_get() like:
if (F_ISSET(S2C(session), WT_CONN_CKPT_METADATA_FAST_PARSER))
    __wt_meta_ckptlist_get_custom()
else {
    current code
}

Reply:
Ok I did what you suggested. Like you said it might make patching a bit trickier.

One thing I was thinking about is to do:

```
if (F_ISSET()) {
    __wt_meta_ckptlist_get_custom();
    goto skip;
}
current code
skip:
   ...
```

This would make the diff cleaner, but is a bit unusual.